[New Integration] Envoyproxy

tehbooom opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

- Enhancement 

## Proposed commit message

This PR adds the envoyproxy integration for logs and metrics. 
<!-- Mandatory
Explain here the changes you made on the PR.

Please explain:

- WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
- WHY:  the rationale/motivation for the changes

This text will be pasted into the squash dialog when the change is committed and will be
a long term historical record of the change to help future contributors understand the
change, please help them by making it clear and comprehensive, they may be you.

If the commit title is adequate to describe both of these things, The text here may be omitted
or replaced with "See title". The title of the PR will be used as the commit message title when
the merge is made and the "See title" marker will be removed if present.

The text here and the PR title will be subject to the PR review process.
-->

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ] 

## How to test this PR locally


## Related issues

- Closes #9383 

## Screenshots

<!-- Optional
Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration
- dashboards with collected metrics or logs
-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16314)
* Commit: 5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16314","number":16314,"commit":"5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad"}],"number":16314}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16320)
* Commit: 37cc117baeae57c81a9f8edecce60c38a37045d3

### Failed CI Steps
* [Check integrations envoyproxy](https://buildkite.com/elastic/integrations/builds/16320#0192206c-066b-4701-89bb-40b3ed3128fd)

### History
* :green_heart: [Build #16314](https://buildkite.com/elastic/integrations/builds/16314) succeeded 5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16320","number":16320,"commit":"37cc117baeae57c81a9f8edecce60c38a37045d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16314","number":16314,"commit":"5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad"}],"number":16320}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16321)
* Commit: b231c55590b674d5c487a7b921c150c725c1c4ca

### History
* :broken_heart: [Build #16320](https://buildkite.com/elastic/integrations/builds/16320) failed 37cc117baeae57c81a9f8edecce60c38a37045d3
* :green_heart: [Build #16314](https://buildkite.com/elastic/integrations/builds/16314) succeeded 5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16321","number":16321,"commit":"b231c55590b674d5c487a7b921c150c725c1c4ca"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16320","number":16320,"commit":"37cc117baeae57c81a9f8edecce60c38a37045d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16314","number":16314,"commit":"5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad"}],"number":16321}
buildkite-pr-comment-->

ritalwar commented on packages/envoyproxy/docs/README.md:
```diff
@@ -0,0 +1,266 @@
+You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
+You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your own hardware.
+
+## Setup
```
Should we add any compatibility details, troubleshooting steps, or common setup information to help users avoid confusion in the future?

tehbooom replied:
I can add what version of envoy it was tested on? For stats and logs datastreams I added how to set those up specifically. Is that sufficient?

ritalwar replied:
Yes, I think we're fine as long as we include enough details.

ritalwar commented on packages/envoyproxy/data_stream/log/agent/stream/log.yml.hbs:
```diff
@@ -0,0 +1,15 @@
+{{/each}}
+{{#contains "forwarded" tags}}
+publisher_pipeline.disable_host: true
+{{/contains}}
```
We might need to add processors like we do in other log datastreams. In that case, we'd include:
```
{{#if processors}}
processors:
{{processors}}
{{/if}}
```

tehbooom replied:
changed to filestream and added the processors

ritalwar commented on packages/envoyproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,99 @@
+---
+description: Pipeline for normalizing envoyproxy logs
+processors:
+- set:
```
We should review whether we need to change the default settings of `ignore_empty_value` and `ignore_failure` from `false` to `true` for all processors in use in all pipelines.

tehbooom replied:
This was copied from our beats pipeline. I dont think its necessary for these to be here as they are mostly for the `timestamp` field and I have yet to see this be a `null` value. Im going to remove these settings

tehbooom replied:
Unsure if i should remove [this](https://github.com/tehbooom/integrations/blob/b42201c501cb0cd720cb08d5abeb7347bb1385c1/packages/envoyproxy/data_stream/log/elasticsearch/ingest_pipeline/json.yml#L13). I think the agent running on k8s will populate the `kubernetes` field. Would we want to overwrite that? Doesnt this `ignore_failure` handle that? I think if it does we can add a conditional to verify its not already set versus ignoring errors

ritalwar replied:
I think ignore_failure should handle. Let's keep it for now.

ritalwar reviewed (commented):
For all newly developed integrations, we enable TSDB by identifying dimensions. You can read more about this process and follow the guidelines for implementation in this integration [here](https://www.elastic.co/guide/en/integrations-developer/current/developer-tsds-guidelines.html).

tehbooom commented on packages/envoyproxy/LICENSE.txt:
```diff
@@ -0,0 +1,93 @@
+Elastic License 2.0
```
removed the license

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16869)
* Commit: b42201c501cb0cd720cb08d5abeb7347bb1385c1

### Failed CI Steps
* [Check integrations envoyproxy](https://buildkite.com/elastic/integrations/builds/16869#0192676c-98a7-4b3c-9363-c783b2bb820b)

### History
* :green_heart: [Build #16321](https://buildkite.com/elastic/integrations/builds/16321) succeeded b231c55590b674d5c487a7b921c150c725c1c4ca
* :broken_heart: [Build #16320](https://buildkite.com/elastic/integrations/builds/16320) failed 37cc117baeae57c81a9f8edecce60c38a37045d3
* :green_heart: [Build #16314](https://buildkite.com/elastic/integrations/builds/16314) succeeded 5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16869","number":16869,"commit":"b42201c501cb0cd720cb08d5abeb7347bb1385c1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16321","number":16321,"commit":"b231c55590b674d5c487a7b921c150c725c1c4ca"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16320","number":16320,"commit":"37cc117baeae57c81a9f8edecce60c38a37045d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16314","number":16314,"commit":"5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad"}],"number":16869}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16875)
* Commit: eadd1879617125a58a975e5f4666c50231f783ce

### Failed CI Steps
* [Check integrations envoyproxy](https://buildkite.com/elastic/integrations/builds/16875#019267f9-1cac-4d09-a63b-3eb5473ded75)

### History
* :broken_heart: [Build #16869](https://buildkite.com/elastic/integrations/builds/16869) failed b42201c501cb0cd720cb08d5abeb7347bb1385c1
* :green_heart: [Build #16321](https://buildkite.com/elastic/integrations/builds/16321) succeeded b231c55590b674d5c487a7b921c150c725c1c4ca
* :broken_heart: [Build #16320](https://buildkite.com/elastic/integrations/builds/16320) failed 37cc117baeae57c81a9f8edecce60c38a37045d3
* :green_heart: [Build #16314](https://buildkite.com/elastic/integrations/builds/16314) succeeded 5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16875","number":16875,"commit":"eadd1879617125a58a975e5f4666c50231f783ce"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16869","number":16869,"commit":"b42201c501cb0cd720cb08d5abeb7347bb1385c1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16321","number":16321,"commit":"b231c55590b674d5c487a7b921c150c725c1c4ca"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16320","number":16320,"commit":"37cc117baeae57c81a9f8edecce60c38a37045d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16314","number":16314,"commit":"5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad"}],"number":16875}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/16909)
* Commit: 8e67a3da55c04c35d1a90a85964e032fb9330cce

### History
* :broken_heart: [Build #16875](https://buildkite.com/elastic/integrations/builds/16875) failed eadd1879617125a58a975e5f4666c50231f783ce
* :broken_heart: [Build #16869](https://buildkite.com/elastic/integrations/builds/16869) failed b42201c501cb0cd720cb08d5abeb7347bb1385c1
* :green_heart: [Build #16321](https://buildkite.com/elastic/integrations/builds/16321) succeeded b231c55590b674d5c487a7b921c150c725c1c4ca
* :broken_heart: [Build #16320](https://buildkite.com/elastic/integrations/builds/16320) failed 37cc117baeae57c81a9f8edecce60c38a37045d3
* :green_heart: [Build #16314](https://buildkite.com/elastic/integrations/builds/16314) succeeded 5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16909","number":16909,"commit":"8e67a3da55c04c35d1a90a85964e032fb9330cce"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16875","number":16875,"commit":"eadd1879617125a58a975e5f4666c50231f783ce"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16869","number":16869,"commit":"b42201c501cb0cd720cb08d5abeb7347bb1385c1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16321","number":16321,"commit":"b231c55590b674d5c487a7b921c150c725c1c4ca"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16320","number":16320,"commit":"37cc117baeae57c81a9f8edecce60c38a37045d3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16314","number":16314,"commit":"5c6dac2f94a3de28e21bcdcc0b098fdfa9de41ad"}],"number":16909}
buildkite-pr-comment-->

ritalwar commented on packages/envoyproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,97 @@
+    value: network
+- set:
+    field: event.ingested
+    value: '{{_ingest.timestamp}}'
```
I recommend adding `ignore_empty_value: true` and `ignore_failure: true` for this field.

ritalwar commented on packages/envoyproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,97 @@
+    field: event.ingested
+    value: '{{_ingest.timestamp}}'
+- pipeline:
+    if: ctx.message.charAt(0) == (char)("{")
```
I think using `if: ctx.message.startsWith('{')` is more cleaner.
```suggestion
    if: ctx.message.startsWith('{')
```

ritalwar commented on packages/envoyproxy/data_stream/log/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,97 @@
+    if: ctx.message.charAt(0) == (char)("{")
+    name: '{{ IngestPipeline "json"}}'
+- pipeline:
+    if: ctx.message.charAt(0) != (char)("{")
```
Can we test if this works:
```suggestion
    if: ctx.message.first_char != '{'
```

ritalwar commented:
> For all newly developed integrations, we enable TSDB by identifying dimensions. You can read more about this process and follow the guidelines for implementation in this integration [here](https://www.elastic.co/guide/en/integrations-developer/current/developer-tsds-guidelines.html).

@tehbooom Did you get a chance to look at this?

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/17449)
* Commit: 0ddce0278d78ac9e74c500e9a508432933fd96ee

### History
* :green_heart: [Build #16967](https://buildkite.com/elastic/integrations/builds/16967) succeeded cf0107cf3750afb1a1e1fd064e44e93901720705
* :green_heart: [Build #16921](https://buildkite.com/elastic/integrations/builds/16921) succeeded 0b15d12828f0955387fb20904ff47b19bb5a3435
* :green_heart: [Build #16909](https://buildkite.com/elastic/integrations/builds/16909) succeeded 8e67a3da55c04c35d1a90a85964e032fb9330cce
* :broken_heart: [Build #16875](https://buildkite.com/elastic/integrations/builds/16875) failed eadd1879617125a58a975e5f4666c50231f783ce
* :broken_heart: [Build #16869](https://buildkite.com/elastic/integrations/builds/16869) failed b42201c501cb0cd720cb08d5abeb7347bb1385c1
* :green_heart: [Build #16321](https://buildkite.com/elastic/integrations/builds/16321) succeeded b231c55590b674d5c487a7b921c150c725c1c4ca

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/17449","number":17449,"commit":"0ddce0278d78ac9e74c500e9a508432933fd96ee"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16967","number":16967,"commit":"cf0107cf3750afb1a1e1fd064e44e93901720705"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16921","number":16921,"commit":"0b15d12828f0955387fb20904ff47b19bb5a3435"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16909","number":16909,"commit":"8e67a3da55c04c35d1a90a85964e032fb9330cce"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16875","number":16875,"commit":"eadd1879617125a58a975e5f4666c50231f783ce"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16869","number":16869,"commit":"b42201c501cb0cd720cb08d5abeb7347bb1385c1"}],"number":17449}
buildkite-pr-comment-->

kush-elastic commented on packages/envoyproxy/data_stream/log/sample_event.json:
```diff
@@ -0,0 +1,99 @@
+    },
+    "envoyproxy": {
+        "log": {
+            "authority": "-",
```
@tehbooom,
Do we know if there are predefined values for `authority`?
can you add processors to identify if `authority` is `-`?
it will be good for users to look at some values like `N/A` instead of just `-`.
WDYT?

tehbooom replied:
I dont think having `N/A` or `-` would make sense in an event. If it doesn't exist in the original event then it shouldn't exist after we process it. I will remove all fields that have a value of `-`

kush-elastic replied:
That would be good. please remove them.

kush-elastic commented on packages/envoyproxy/data_stream/log/sample_event.json:
```diff
@@ -0,0 +1,99 @@
+        "original": "ACCESS [2019-04-06T06:20:05.972Z] \"- - -\" 0 UF,URX 0 0 0 - \"-\" \"-\" \"-\" \"-\" \"127.0.0.1:9200\"",
+        "outcome": "failure",
+        "type": [
+            "connection",
+            "connection"
```
are these duplicate values needed here?
can we somehow remove duplicate values using pipeline?

kush-elastic commented on packages/envoyproxy/data_stream/stats/elasticsearch/ingest_pipeline/default.yml:
```diff
@@ -0,0 +1,239 @@
+- dot_expander:
+    field: "*"
+    path: labels
+- rename:
+    field: labels.envoy.cluster_name
+    target_field: envoyproxy.cluster_name
+    ignore_missing: true
```
can we not just use single rename processor to just rename `labels.envoy` to `envoyproxy` here?
```suggestion
- rename:
    field: labels.envoy
    target_field: envoyproxy
    ignore_missing: true
```

tehbooom replied:
Tried this and some events could not be indexed.

kush-elastic commented on packages/envoyproxy/data_stream/stats/fields/fields.yml:
```diff
@@ -0,0 +1,265 @@
+- name: envoyproxy
+  type: group
+  fields:
+    - name: cluster_name
```
does envoyproxy not have any single unique identifier that we can use for dimension?

tehbooom replied:
No it doesn't seem like it to me or at least obvious to me. Envoy has a list of tags [here](https://github.com/envoyproxy/envoy/blob/main/source/common/config/well_known_names.h#L87). Each event from statsd has a combination of these tags. Most often it is `envoy.cluster_name` in the event with another `envoy.cluster.X`. For example in our `envoy-stats.log` out of 200 events 114 had this tag. I'm not sure if we want to include only `envoy.cluster.X` in TSDS

kush-elastic replied:
Got it.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/18330)
* Commit: f619f1d479c66787627b96ee3b40b29c2217e0b0

### History
* :green_heart: [Build #17449](https://buildkite.com/elastic/integrations/builds/17449) succeeded 0ddce0278d78ac9e74c500e9a508432933fd96ee
* :green_heart: [Build #16967](https://buildkite.com/elastic/integrations/builds/16967) succeeded cf0107cf3750afb1a1e1fd064e44e93901720705
* :green_heart: [Build #16921](https://buildkite.com/elastic/integrations/builds/16921) succeeded 0b15d12828f0955387fb20904ff47b19bb5a3435
* :green_heart: [Build #16909](https://buildkite.com/elastic/integrations/builds/16909) succeeded 8e67a3da55c04c35d1a90a85964e032fb9330cce
* :broken_heart: [Build #16875](https://buildkite.com/elastic/integrations/builds/16875) failed eadd1879617125a58a975e5f4666c50231f783ce
* :broken_heart: [Build #16869](https://buildkite.com/elastic/integrations/builds/16869) failed b42201c501cb0cd720cb08d5abeb7347bb1385c1

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18330","number":18330,"commit":"f619f1d479c66787627b96ee3b40b29c2217e0b0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/17449","number":17449,"commit":"0ddce0278d78ac9e74c500e9a508432933fd96ee"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16967","number":16967,"commit":"cf0107cf3750afb1a1e1fd064e44e93901720705"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16921","number":16921,"commit":"0b15d12828f0955387fb20904ff47b19bb5a3435"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16909","number":16909,"commit":"8e67a3da55c04c35d1a90a85964e032fb9330cce"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16875","number":16875,"commit":"eadd1879617125a58a975e5f4666c50231f783ce"}],"number":18330}
buildkite-pr-comment-->

kush-elastic reviewed (approved):
LGTM!

ritalwar commented on packages/envoyproxy/data_stream/stats/fields/fields.yml:
```diff
@@ -0,0 +1,265 @@
+    - name: cluster_name
+      type: keyword
+      description: Cluster name tag
+      # Reason to add as a dimension field: Cluster name tag
```
The reasons provided for selecting all dimensions seem to mirror the field descriptions. Could we instead include the actual rationale for choosing them as dimensions? This would offer clearer context for future reference.

tehbooom replied:
Added a more detailed reason for each dimension. Let me know if this is sufficient.

ritalwar commented:
Did we not create any default dashboards for this package?

tehbooom commented:
> Did we not create any default dashboards for this package?

I did not create any dashboards for this. Do we need to for the initial package or can this be implemented in another PR? @ritalwar

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/18607)
* Commit: 250d9ef238044f6f1c4149b15b446369f3e9f3ce

### History
* :green_heart: [Build #18330](https://buildkite.com/elastic/integrations/builds/18330) succeeded f619f1d479c66787627b96ee3b40b29c2217e0b0
* :green_heart: [Build #17449](https://buildkite.com/elastic/integrations/builds/17449) succeeded 0ddce0278d78ac9e74c500e9a508432933fd96ee
* :green_heart: [Build #16967](https://buildkite.com/elastic/integrations/builds/16967) succeeded cf0107cf3750afb1a1e1fd064e44e93901720705
* :green_heart: [Build #16921](https://buildkite.com/elastic/integrations/builds/16921) succeeded 0b15d12828f0955387fb20904ff47b19bb5a3435
* :green_heart: [Build #16909](https://buildkite.com/elastic/integrations/builds/16909) succeeded 8e67a3da55c04c35d1a90a85964e032fb9330cce
* :broken_heart: [Build #16875](https://buildkite.com/elastic/integrations/builds/16875) failed eadd1879617125a58a975e5f4666c50231f783ce

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18607","number":18607,"commit":"250d9ef238044f6f1c4149b15b446369f3e9f3ce"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18330","number":18330,"commit":"f619f1d479c66787627b96ee3b40b29c2217e0b0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/17449","number":17449,"commit":"0ddce0278d78ac9e74c500e9a508432933fd96ee"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16967","number":16967,"commit":"cf0107cf3750afb1a1e1fd064e44e93901720705"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16921","number":16921,"commit":"0b15d12828f0955387fb20904ff47b19bb5a3435"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16909","number":16909,"commit":"8e67a3da55c04c35d1a90a85964e032fb9330cce"}],"number":18607}
buildkite-pr-comment-->

agithomas commented:
@tehbooom , Limiting the number of dimensions is always best, to avoid extremely high cardinality.  The fewer numbers of dimensions are always better. 


Not all fields with the type `keyword` must be mapped as a `dimension`.The target must be to identify the set of (dimension-mapped) fields that uniquely identify an event in the ES at a given timestamp. Usually, the unique attributes of the resources that are being monitored makes this list. 

You may run the [TSDB Migration toolkit](https://github.com/elastic/TSDB-migration-test-kit) to identify and incrementally add the dimension fields.

You may also check - if any of the [common dimension fields](https://www.elastic.co/guide/en/integrations-developer/current/developer-tsds-guidelines.html#_ecs_fields) to be included as part of the dimension mapping.

Also, `log` based `dataset` fields must be excluded from `dimension` mapping.

tehbooom commented:
@agithomas , I removed a big chunk of dimensions fields and only kept those that seem like they wont change as much over time. I also added `container.id` as a dimension. 

Should we still create a dashboard for the initial integration or can this be for another PR? 

Finally, any thoughts on the following fields also being added as dimensions?

- host.name
- service.address
- cloud.account.id
- cloud.provider
- cloud.region
- cloud.availability_zone
- agent.id
- cloud.instance.id

agithomas commented:
> Finally, any thoughts on the following fields also being added as dimensions?
In the absence of a globally unique identifier, the common dimension fields could help. 

For example, if envoyproxy has to be installed in multiple regions, does there exist a unique identifier that separates the installation from the first region from the second region?  Adding the above-mentioned dimension fields helps if there exist no Globally unique resource identifier.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/18770)
* Commit: 144448cef3eaf479afcbac2060ef5319d5f454e3

### History
* :green_heart: [Build #18607](https://buildkite.com/elastic/integrations/builds/18607) succeeded 250d9ef238044f6f1c4149b15b446369f3e9f3ce
* :green_heart: [Build #18330](https://buildkite.com/elastic/integrations/builds/18330) succeeded f619f1d479c66787627b96ee3b40b29c2217e0b0
* :green_heart: [Build #17449](https://buildkite.com/elastic/integrations/builds/17449) succeeded 0ddce0278d78ac9e74c500e9a508432933fd96ee
* :green_heart: [Build #16967](https://buildkite.com/elastic/integrations/builds/16967) succeeded cf0107cf3750afb1a1e1fd064e44e93901720705
* :green_heart: [Build #16921](https://buildkite.com/elastic/integrations/builds/16921) succeeded 0b15d12828f0955387fb20904ff47b19bb5a3435
* :green_heart: [Build #16909](https://buildkite.com/elastic/integrations/builds/16909) succeeded 8e67a3da55c04c35d1a90a85964e032fb9330cce

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18770","number":18770,"commit":"144448cef3eaf479afcbac2060ef5319d5f454e3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18607","number":18607,"commit":"250d9ef238044f6f1c4149b15b446369f3e9f3ce"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18330","number":18330,"commit":"f619f1d479c66787627b96ee3b40b29c2217e0b0"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/17449","number":17449,"commit":"0ddce0278d78ac9e74c500e9a508432933fd96ee"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16967","number":16967,"commit":"cf0107cf3750afb1a1e1fd064e44e93901720705"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/16921","number":16921,"commit":"0b15d12828f0955387fb20904ff47b19bb5a3435"}],"number":18770}
buildkite-pr-comment-->

agithomas commented:
> Should we still create a dashboard for the initial integration or can this be for another PR?
Unless this is an urgent requirement, it is always good to package the dashboard as part of the initial release.

tehbooom commented:
Verified this integration with envoyproxy version 1.32.1.

Also tested with [TSDB Migration toolkit](https://github.com/elastic/TSDB-migration-test-kit) with results below

```
Values being used:
        data_stream = metrics-envoyproxy.stats-76262
        docs_index = First index of the data stream
        settings_mappings_index = Last index of the data stream
        max_docs = All documents
        get_overlapping_files = True
        directory_overlapping_files = overwritten-docs-metrics-envoyproxy.stats-76262
        display_docs = 10
        copy_docs_per_dimension = 2

You're testing with version 8.16.0.

Testing data stream metrics-envoyproxy.stats-76262.
Index being used for the documents is .ds-metrics-envoyproxy.stats-76262-2024.11.27-000001.
Index being used for the settings and mappings is .ds-metrics-envoyproxy.stats-76262-2024.11.27-000001.

The time series fields for the TSDB index are:
        - dimension (4 fields):
                - container.id
                - envoyproxy.cluster_name
                - envoyproxy.http_conn_manager_prefix
                - envoyproxy.listener_address
        - counter (52 fields):
                - envoyproxy.envoy_cluster_default_total_match_count.count
                - envoyproxy.envoy_cluster_external_upstream_rq.count
                - envoyproxy.envoy_cluster_external_upstream_rq_completed.count
                - envoyproxy.envoy_cluster_external_upstream_rq_xx.count
                - envoyproxy.envoy_cluster_health_check_attempt.count
                - envoyproxy.envoy_cluster_health_check_failure.count
                - envoyproxy.envoy_cluster_health_check_network_failure.count
                - envoyproxy.envoy_cluster_membership_change.count
                - envoyproxy.envoy_cluster_ssl_ciphers.count
                - envoyproxy.envoy_cluster_ssl_curves.count
                - envoyproxy.envoy_cluster_ssl_handshake.count
                - envoyproxy.envoy_cluster_ssl_sigalgs.count
                - envoyproxy.envoy_cluster_ssl_versions.count
                - envoyproxy.envoy_cluster_upstream_cx_destroy.count
                - envoyproxy.envoy_cluster_upstream_cx_destroy_local.count
                - envoyproxy.envoy_cluster_upstream_cx_destroy_local_with_active_rq.count
                - envoyproxy.envoy_cluster_upstream_cx_destroy_with_active_rq.count
                - envoyproxy.envoy_cluster_upstream_cx_rx_bytes_total.count
                - envoyproxy.envoy_cluster_upstream_cx_tx_bytes_total.count
                - envoyproxy.envoy_cluster_upstream_rq.count
                - envoyproxy.envoy_cluster_upstream_rq_pending_total.count
                - envoyproxy.envoy_cluster_upstream_rq_total.count
                - envoyproxy.envoy_http_downstream_cx_destroy.count
                - envoyproxy.envoy_http_downstream_cx_destroy_active_rq.count
                - envoyproxy.envoy_http_downstream_cx_destroy_local.count
                - envoyproxy.envoy_http_downstream_cx_destroy_remote.count
                - envoyproxy.envoy_http_downstream_cx_destroy_remote_active_rq.count
                - envoyproxy.envoy_http_downstream_cx_http1_total.count
                - envoyproxy.envoy_http_downstream_cx_rx_bytes_total.count
                - envoyproxy.envoy_http_downstream_cx_ssl_total.count
                - envoyproxy.envoy_http_downstream_cx_total.count
                - envoyproxy.envoy_http_downstream_cx_tx_bytes_total.count
                - envoyproxy.envoy_http_downstream_rq_completed.count
                - envoyproxy.envoy_http_downstream_rq_http1_total.count
                - envoyproxy.envoy_http_downstream_rq_rx_reset.count
                - envoyproxy.envoy_http_downstream_rq_time.count
                - envoyproxy.envoy_http_downstream_rq_total.count
                - envoyproxy.envoy_http_downstream_rq_xx.count
                - envoyproxy.envoy_http_rq_direct_response.count
                - envoyproxy.envoy_http_rq_total.count
                - envoyproxy.envoy_listener_connections_accepted_per_socket_event.count
                - envoyproxy.envoy_listener_downstream_cx_destroy.count
                - envoyproxy.envoy_listener_downstream_cx_total.count
                - envoyproxy.envoy_listener_http_downstream_rq_completed.count
                - envoyproxy.envoy_listener_http_downstream_rq_xx.count
                - envoyproxy.envoy_listener_ssl_ciphers.count
                - envoyproxy.envoy_listener_ssl_curves.count
                - envoyproxy.envoy_listener_ssl_handshake.count
                - envoyproxy.envoy_listener_ssl_no_certificate.count
                - envoyproxy.envoy_listener_ssl_ocsp_staple_requests.count
                - envoyproxy.envoy_listener_ssl_versions.count
                - envoyproxy.envoy_listener_worker_downstream_cx_total.count
        - gauge (15 fields):
                - envoyproxy.envoy_cluster_circuit_breakers_default_cx_open.value
                - envoyproxy.envoy_cluster_membership_degraded.value
                - envoyproxy.envoy_cluster_membership_excluded.value
                - envoyproxy.envoy_cluster_membership_healthy.value
                - envoyproxy.envoy_cluster_membership_total.value
                - envoyproxy.envoy_cluster_upstream_cx_active.value
                - envoyproxy.envoy_cluster_warming_state.value
                - envoyproxy.envoy_http_downstream_cx_active.value
                - envoyproxy.envoy_http_downstream_cx_http1_active.value
                - envoyproxy.envoy_http_downstream_cx_rx_bytes_buffered.value
                - envoyproxy.envoy_http_downstream_cx_ssl_active.value
                - envoyproxy.envoy_http_downstream_rq_active.value
                - envoyproxy.envoy_listener_downstream_cx_active.value
                - envoyproxy.envoy_listener_downstream_pre_cx_active.value
                - envoyproxy.envoy_listener_worker_downstream_cx_active.value
        - routing_path (4 fields):
                - container.id
                - envoyproxy.cluster_name
                - envoyproxy.http_conn_manager_prefix
                - envoyproxy.listener_address

Index tsdb-index-enabled successfully created.

Copying documents from .ds-metrics-envoyproxy.stats-76262-2024.11.27-000001 to tsdb-index-enabled...
All 224 documents taken from index .ds-metrics-envoyproxy.stats-76262-2024.11.27-000001 were successfully placed to index tsdb-index-enabled.
```

I think we will also need to remove the beats from the package registry as seen below

![Screenshot From 2024-11-27 15-53-04](https://github.com/user-attachments/assets/8da8ae96-3b02-49cd-a280-24ea0366c86f)

ritalwar reviewed (approved):
Please handle TSDB-related feedback. The rest looks good, and we can create dashboards in a separate PR.

agithomas commented on packages/envoyproxy/data_stream/stats/fields/fields.yml:
```diff
@@ -0,0 +1,183 @@
+    - name: mongo_prefix
+      type: keyword
+      description: Stats prefix for the Mongo Proxy network filter
+    - name: mongo_cmd
```
Could there be multiple values for this field ? If so, kindly make this field as a dimension.

agithomas replied:
For all the filters mentioned below, including GRPC, mongo, Router HTTP, Network , RBAC network filter, etc, kindly verify if multiple values are possible at a given time. If so, it may be added to the dimensions list. Please confirm.

agithomas commented on packages/envoyproxy/data_stream/stats/fields/fields.yml:
```diff
@@ -0,0 +1,183 @@
+    - name: mongo_cmd
+      type: keyword
+      description: Request command for the Mongo Proxy network filter
+    - name: mongo_collection
```
Could there be multiple values for this field ? If so, kindly make this field as a dimension.

agithomas commented on packages/envoyproxy/data_stream/stats/fields/fields.yml:
```diff
@@ -0,0 +1,183 @@
+    - name: ext_authz_prefix
+      type: keyword
+      description: Stats prefix for the ext_authz HTTP filter
+    - name: worker_id
```
Can there be multiple workers for a given listener, the stats / metrics of these workers are captured separately. If so, please add this field to the dimension list.

tehbooom replied:
Added to dimensions

tehbooom commented on packages/envoyproxy/data_stream/stats/sample_event.json:
```diff
@@ -0,0 +1,63 @@
+        "snapshot": false,
+        "version": "8.15.1"
+    },
+    "envoyproxy": {
```
The existing beats metrics does not use statsd

tehbooom replied:
Renamed to `envoyproxy.*` to `envoy.*`

agithomas commented:
@daniela-elastic , as discussed kindly inspect the metrics included as part of the integration. Please verify if are not having more metrics than actually recommended.

Secondly, please review the package ownership details.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/19554)
* Commit: 960e368635eca6c80f0a96d80212420dd7c0efaf

### History
* :green_heart: [Build #18962](https://buildkite.com/elastic/integrations/builds/18962) succeeded dfc309966d765e974d6881dcfae9f39dada0e5eb
* :green_heart: [Build #18809](https://buildkite.com/elastic/integrations/builds/18809) succeeded 1378a29e80f1d90bfade2409011ca95d95acf83e
* :green_heart: [Build #18770](https://buildkite.com/elastic/integrations/builds/18770) succeeded 144448cef3eaf479afcbac2060ef5319d5f454e3
* :green_heart: [Build #18607](https://buildkite.com/elastic/integrations/builds/18607) succeeded 250d9ef238044f6f1c4149b15b446369f3e9f3ce
* :green_heart: [Build #18330](https://buildkite.com/elastic/integrations/builds/18330) succeeded f619f1d479c66787627b96ee3b40b29c2217e0b0
* :green_heart: [Build #17449](https://buildkite.com/elastic/integrations/builds/17449) succeeded 0ddce0278d78ac9e74c500e9a508432933fd96ee

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/19554","number":19554,"commit":"960e368635eca6c80f0a96d80212420dd7c0efaf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18962","number":18962,"commit":"dfc309966d765e974d6881dcfae9f39dada0e5eb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18809","number":18809,"commit":"1378a29e80f1d90bfade2409011ca95d95acf83e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18770","number":18770,"commit":"144448cef3eaf479afcbac2060ef5319d5f454e3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18607","number":18607,"commit":"250d9ef238044f6f1c4149b15b446369f3e9f3ce"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/18330","number":18330,"commit":"f619f1d479c66787627b96ee3b40b29c2217e0b0"}],"number":19554}
buildkite-pr-comment-->

agithomas commented:
Hi @lalit-satapathy and @daniela-elastic 

Could you please look at this comment](https://github.com/elastic/integrations/pull/11215#issuecomment-2518137320) and advice? 

Thanks

lalit-satapathy commented:
> Secondly, please review the package ownership details.
LGTM.
